owncloudApp/src/main/java/com/owncloud/android/workers/UploadFileFromFileSystemWorker.kt

```diff
@@ -119,24 +121,34 @@ class UploadFileFromFileSystemWorker(
      * We will ask for thumbnails after the upload
      * We will update info about the file (modification timestamp and etag)
      */
-    private fun updateFilesDatabaseWithNewEtag() {
+    private fun updateFilesDatabaseWithLatestDetails() {
         val currentTime = System.currentTimeMillis()
-        if (ocTransfer.forceOverwrite) {
-            val getFileByRemotePathUseCase: GetFileByRemotePathUseCase by inject()
-            val file = getFileByRemotePathUseCase.execute(GetFileByRemotePathUseCase.Params(account.name, ocTransfer.remotePath))
-            file.getDataOrNull()?.let { ocFile ->
-                ocFile.copy(
-                    needsToUpdateThumbnail = true,
-                    etag = uploadFileOperation.etag,
-                    length = (File(ocTransfer.localPath).length()),
-                    lastSyncDateForData = currentTime,
-                    modifiedAtLastSyncForData = currentTime,
-                    etagInConflict = null
-                ).let {
-                    saveFileOrFolderUseCase.execute(SaveFileOrFolderUseCase.Params(it))
+        val getFileByRemotePathUseCase: GetFileByRemotePathUseCase by inject()
+        val file = getFileByRemotePathUseCase.execute(GetFileByRemotePathUseCase.Params(account.name, ocTransfer.remotePath))
+        file.getDataOrNull()?.let { ocFile ->
+            val fileWithNewDetails =
+                if (ocTransfer.forceOverwrite) {
+                    ocFile.copy(
+                        needsToUpdateThumbnail = true,
+                        etag = uploadFileOperation.etag,
+                        length = (File(ocTransfer.localPath).length()),
+                        lastSyncDateForData = currentTime,
+                        modifiedAtLastSyncForData = currentTime,
+                    )
+                } else {
+                    // Uploading a file should remove any conflicts on the file.
+                    ocFile.copy(
+                        etagInConflict = null,
```

Comment:
It will be cleaned on the next cleanConflictUseCase, so I think this is not needed at all. 👍

Reply:
My bad, I removed the `storagePath` instead of the `etagInConflict` field. Fixed!! 👍